pycbc/events/cuts.py

```diff
@@ -0,0 +1,336 @@
+# with this program; if not, write to the Free Software Foundation, Inc.,
+# 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+
+#
+# =============================================================================
+#
+#                                   Preamble
+#
+# =============================================================================
+#
+
+"""
+This module contains functions for reading in command line options and
+applying cuts to triggers or templates in the offline search
+"""
+import logging
+import copy
+import numpy as np
+from pycbc.events import ranking
+from pycbc.io import hdf
+from pycbc.bank import conversions as bank_conv
+from pycbc.io import get_chisq_from_file_choice
+
+sngl_rank_keys = ranking.sngls_ranking_function_dict.keys()
+
+trigger_param_choices = list(sngl_rank_keys)
+trigger_param_choices += [cc + '_chisq' for cc in hdf.chisq_choices]
+trigger_param_choices += ['end_time', 'psd_var_val', 'sigmasq']
+
+template_fit_param_choices = ['fit_by_fit_coeff', 'smoothed_fit_coeff',
+                              'fit_by_count_above_thresh',
+                              'smoothed_fit_count_above_thresh',
+                              'fit_by_count_in_template',
+                              'smoothed_fit_count_in_template']
+template_param_choices = bank_conv.conversion_options + \
+                             template_fit_param_choices
+
+ineq_choices = ['upper', 'lower', 'upper_inc', 'lower_inc']
+
+
+def insert_cuts_option_group(parser):
+    """
+    Add options to the parser for cuts to the templates/triggers
+    """
+    parser.add_argument('--trigger-cuts', nargs='+',
+                        help="Cuts to apply to the triggers, supplied as "
+                             "PARAMETER:VALUE:LIMIT, where, PARAMETER is the "
+                             "parameter to be cut, VALUE is the value at "
+                             "which it is cut, which must be convertible to "
+                             "a float, and LIMIT is one of '"
+                             + "', '".join(ineq_choices) +
+                             "' to indicate the inequality needed. "
+                             "PARAMETER is one of:'"
+                             + "', '".join(trigger_param_choices) +
+                             "'. For example snr:6:LOWER places the "
+                             "requirement that matched filter SNR must be > 6")
+    parser.add_argument('--template-cuts', nargs='+',
+                        help="Cuts to apply to the triggers, supplied as "
+                             "PARAMETER:VALUE:LIMIT. Format is the same as in "
+                             "--trigger-cuts. PARAMETER can be one of '"
+                             + "', '".join(template_param_choices) + "'.")
+
+
+# What are the inequalities associated with the cuts?
+# 'upper' means upper limit, and so requires value < the threshold
+ineq_functions = {
+    'upper': np.less,
```

Comment:
Again, it's not totally clear if triggers for which the inequality is true are kept or cut.  It looks like they are kept.  So 'upper' means apply a cut at the top end of the range, etc. 
Then I'd suggest '.. requires value < threshold to keep a trigger'